### DISCUSSION TITLE
Rejected invalid OpenID token: JWT processing failed

### DISCUSSION THREAD

**Discussion by DimonW88** [keycloak/keycloak#28308] [Q&A]
Good afternoon, please tell me, I have encountered such a problem. I use docker, launched 3 containers:
nginx proxy mgr
keyclock 21.1
guacamole 1.5.4

In nginx add proxy host:
**auth.mydomain.local to keyclock** 

_custom cfg:
proxy_buffer_size   128k; 
proxy_buffers   4 256k;  
proxy_busy_buffers_size   256k;_

**guac.mydomain.local to guacamole**
I didn't add anything. 

Added realm "guacamole" and client "guacamole" to keycloack
**keycloack.docker-compose.yml**
 _environment:
      KC_HTTP_ENABLED: "true"
      KC_HOSTNAME_STRICT: "false"
      KC_HOSTNAME_STRICT_HTTPS: "false"
      KC_HTTP_RELATIVE_PATH: "/auth"
      KEYCLOAK_FRONTEND_URL: "https://auth.mydomain.local/auth/"
      KC_PROXY_ADDRESS_FORWARDING: "true"
      KC_PROXY-HEADERS: "forwarded|xforwarded"_

**guacamole.docker-compose.yml**
 _environment:
      EXTENSIONS: auth-sso-openid
      OPENID_AUTHORIZATION_ENDPOINT: "https://auth.mydomain.local/auth/realms/guacamole/protocol/openid-connect/auth/"
      OPENID_ISSUER: "https://auth.mydomain.local/auth/realms/guacamole"
      OPENID_JWKS_ENDPOINT: "https://auth.mydomain.local/auth/realms/guacamole/protocol/openid-connect/certs"
      OPENID_CLIENT_ID: "guacamole"
      OPENID_REDIRECT_URI: "https://guac.mydomain.local/"_

with these settings, I get an error:
 Rejected invalid OpenID token: JWT processing failed. Additional details: [[17] Unable to process JOSE object
 Unable to find a suitable verification key for JWS w/ header {"alg":"RS256","typ" : "JWT","kid" 

if i change

**guacamole.docker-compose.yml**
 _environment:
      EXTENSIONS: auth-sso-openid
      OPENID_AUTHORIZATION_ENDPOINT: "https://LOCAL-IP-SERVER/auth/realms/guacamole/protocol/openid-connect/auth/"
      OPENID_ISSUER: "https://LOCAL-IP-SERVER/auth/realms/guacamole"
      OPENID_JWKS_ENDPOINT: "https://LOCAL-IP-SERVER/auth/realms/guacamole/protocol/openid-connect/certs"
      OPENID_CLIENT_ID: "guacamole"
      OPENID_REDIRECT_URI: "https://guac.mydomain.local/"_

Then I pass the authorization successfully and everything works great.
Do I understand correctly that nginx breaks JWT? 
maybe you need to add some more parameters to Nginx custom conf so that this does not happen?

**Comment by codespearhead** [comment]
Is this reproducible in the latest major release of Keycloak? If so, please give um an [MRE](https://en.wikipedia.org/wiki/Minimal_reproducible_example#:~:text=In%20computing%2C%20a%20minimal%20reproducible,to%20be%20demonstrated%20and%20reproduced.) in the form of a docker-compose.yml file ?

**Reply by DimonW88** [reply]
I think the problem is not Keycloak. I've tried different releases ,and  another product, it looks very much like nginx is breaking  JWT

**Reply by codespearhead** [reply]
It seems that NginX is stripping some headers, but that's not the default behavior. I don't seem to be able to reproduce your issue with the latest version of Keycloak and NginX.

By the way, [as per the documentation](https://www.keycloak.org/server/reverseproxy): it is `KC_PROXY_HEADERS` not `KC_PROXY-HEADERS`.

Try changing it to: 

```bash
KC_PROXY_HEADERS=forwarded
```
